human-forecasts/server-distribution-code.R

```diff
@@ -0,0 +1,432 @@
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]),
+                                                    sd = as.numeric(rv$width[i])))
+      
+    } else if (input$distribution == "cubic-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]) ^ (1 / 3),
+                                                    sd = as.numeric(rv$width[i]))) ^ 3
+    } else if (input$distribution == "fifth-power-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]) ^ (1 / 5),
+                                                    sd = as.numeric(rv$width[i]))) ^ 5
+    } else if (input$distribution == "seventh-power-normal") {
+      rv[[paste0("forecasts_week_", i)]] <<- (qnorm(quantile_grid, 
+                                                    mean = (rv$median[i]) ^ (1 / 7),
+                                                    sd = as.numeric(rv$width[i]))) ^ 7
+    } 
+    
+    
+  }
+}
+
+
+
+
+output$p_distr <- renderPlotly({
+  circles_pred <- map2(.x = x_pred(), .y  = rv$median, 
+                       ~list(type = "circle",
+                             # anchor circles at (mpg, wt)
+                             xanchor = .x,
+                             yanchor = .y,
+                             # give each circle a 2 pixel diameter
+                             x0 = -5, x1 = 5,
+                             y0 = -5, y1 = 5,
+                             xsizemode = "pixel", 
+                             ysizemode = "pixel",
+                             # other visual properties
+                             fillcolor = "green",
+                             line = list(color = "transparent"))
+  )
+  
+  
+  
+  p <- plot_ly() %>%
+    add_trace(x = df()$target_end_date,
+              y = df()$value, type = "scatter",
+              name = 'observed data',mode = 'lines+markers') %>%     
+    add_trace(x = x_pred(),
+              y = rv$median, type = "scatter",
+              name = 'median prediction',mode = 'lines+markers', color = I("dark green")) %>%
+    layout(title = paste(input$selection), list(
+      xanchor = "left"
+    )) %>%
+    layout(xaxis = list(range = c(min(x()), max(x_pred()) + 5))) %>%
+    layout(yaxis = list(hoverformat = '0f', rangemode = "tozero")) %>%
+    layout(shapes = c(circles_pred)) %>%
+    layout(legend = list(orientation = 'h')) %>%
+    config(edits = list(shapePosition = TRUE)) 
+  
+  if (input$plotscale == "log") {
+    p <- layout(p, yaxis = list(type = "log"))
+  }
+  
+  
+  for (i in prediction_intervals()) {
+    
+    lower_quantile <- round((100 - i) / (2 * 100), 3)
+    upper_quantile <- 1 - lower_quantile
+    
+    
+    lower_bound <- c(rv$forecasts_week_1[round(quantile_grid, 3) == lower_quantile], 
```

Comment:
Helper fn here might be nice

Reply:
so basically something like `get_bound("lower")` and then have the code somewhere else? But I would still have the more or less same code somewhere, I just would call the wrapper here?